### PR TITLE
feat(module:core): add universal logger funcs and deprecation warnings

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- ~~[ ] Tests for the changes have been added (for bug fixes / features)~~
- ~~[ ] Docs have been added / updated (for bug fixes / features)~~

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[x] Other... Please describe: modify the `LoggerServive` to only report warnings and errors in dev mode.
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "no-arg": true,
     "no-bitwise": false,
     "no-consecutive-blank-lines": [true],
-    "no-console": [true, "debug", "info", "time", "timeEnd", "trace"],
+    "no-console": false,
```
Comment (hsuanxyz) [inline tslint.json]:
```suggestion
    "no-console": [true, "debug", "info", "time", "timeEnd", "trace"],
```

在使用的地方 tslint:disable 就行, 否则 lint 无法发现其他地方是否误用

File: tslint.json [hunk 2]
```diff
@@ -84,7 +84,7 @@
     "no-irregular-whitespace": true,
     "no-magic-numbers": false,
     "no-misused-new": true,
-    "no-namespace": [true, "allow-declarations"],
+    "no-namespace": false,
```
Comment (hsuanxyz) [inline tslint.json]:
```suggestion
    "no-namespace": [true, "allow-declarations"],
```
namespace 就是 namespace，不要当模块使用。

File: components/core/logger/logger.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+// tslint:disable:no-any
+
+import { isDevMode } from '@angular/core';
+
+const record: Record<string, boolean> = {};
+
+export function getNzConsolePrefix(): string {
```
Comment (wenqi73) [inline components/core/logger/logger.ts]: 感觉写成一个常量好一点？`NZ_CONSOLE_PREFIX`

Comment (wendzhue) [reply components/core/logger/logger.ts]: 感觉没啥区别

File: components/upload/upload.spec.ts [hunk 4]
```diff
@@ -1233,9 +1233,9 @@ describe('upload', () => {
         expect(comp.options.customRequest).toHaveBeenCalled();
       });
 
-      it('should be warn "Must return Subscription type in [nzCustomRequest] property"', () => {
+      xit('should be warn "Must return Subscription type in [nzCustomRequest] property"', () => {
```
Comment (vthinkxie) [inline components/upload/upload.spec.ts]: any reason use xit?

**Pull request merged**